pkg/cache/scheduler/tas_flavor_snapshot.go

```diff
@@ -1385,6 +1448,34 @@ func (s *TASFlavorSnapshot) fillInCounts(
 	}
 }
 
+// limitingResource finds the resource that is most constraining for scheduling.
+// It compares the remaining capacity for each resource to the requested amount
+// and returns the one with the lowest capacity/request ratio (ties broken by
+// resource name for determinism).
+func limitingResource(requests resources.Requests, capacity resources.Requests, resourceNames []corev1.ResourceName) (corev1.ResourceName, bool) {
```

Comment:
This is actually generalizing the CountIn, so instead of duplicating I would propose calling it `CountInWithLimitingResource`, and then make trivial implemation of CountIn by delegating to that function: 
```golang
func CountIn(request, total Resources) int {
    count, _ := CountInWithLimitingResource(request, total)
    return count
}
```

Reply:
Yup, fixed.